eval/crypt.go

```diff
@@ -150,6 +151,12 @@ func DecryptSecrets(ctx context.Context, filename string, source []byte, decrypt
 			return n, nil, nil
 		}
 
+		// Don't attempt to decrypt secrets outside of valid top-level keys
+		path := n.Syntax().Path()
+		if !(strings.HasPrefix(path, "values.") || strings.HasPrefix(path, "imports.")) {
```

Comment:
@pgavlin thoughts on this approach? another approach is to change where we start walking through the nodes (only walking through `values/imports`) but that's more involved since that logic is shared between encrypt/decrypt

Reply:
I think that this is okay. I don't love that we're using a string for the path here--ideally the path would be stored in a parsed form--but this is essentially what I'd suggest.

The other similar option is to change `rewriteYAML` so that it passes the current path to the visitor. Something like

```
func rewriteYAML(
	ctx context.Context,
	filename string,
	source []byte,
	visitor func(path []any, n syntax.Node) (syntax.Node, syntax.Diagnostics, error),
) ([]byte, error) {
...
}
```

But that would also require changing `syntax.Walk`. So I think what you have is fine.

I'd also go a little further and only rewrite secrets under `values`.